content/personal/telepathicgrunt/open-source-licenses.md

```diff
@@ -0,0 +1,51 @@
+
+- **LGPLv3:** <https://opensource.org/license/lgpl-3-0> : <https://www.tldrlegal.com/license/gnu-lesser-general-public-license-v3-lgpl-3>
+
+  - This license is more restrictive where it requires modders taking your code to make their own project be Open-Source as well using a license compatible with LGPLv3 along with stating where the original code came from (usually a link to original repository). It also requires that the modder states all the changes they did to the copied code. The modder can use the code in a commercial project as long as that project is now Open-Source. Note, there is a catch with LGPLv3 where a server owner could modify a serverside LGPLv3 mod and does not have to reveal the source due to the wording of the license.
+
+- **OSL-3.0:** <https://opensource.org/license/osl-3-0-php> : <https://www.tldrlegal.com/license/open-software-licence-3-0>
+
+  - This one is more relaxed than LGPLv3 where copiers have to Open-Source their projects and give credit to the original code owner but does NOT need to add a link to the original repository nor state what changes were done. The copied code can be used in commercial Open-Source projects as well. A big difference with OSL-3.0 is that it closes the loophole where if a server owner modifies a serverside mod, they now must provide the source when asked.
+
+- **Apache-2.0:** <https://opensource.org/license/apache-2-0> : <https://www.tldrlegal.com/license/apache-license-2-0-apache-2-0>
+
+  - Modders taking your code must add a NOTICED file with attribution to you and state all the changes they did to the copied code. They do not need to Open-Source their mod and can use it in commercial ARR (All Rights Reserved) projects. It is similar to the MIT license but requires a more visible credit to the original owner of the copied code. Apache-2.0 also specifically allows “private use” so a server owner can modify your mod to use on a server without restriction.
+
+- **MPL-2.0:** <https://opensource.org/license/mpl-2-0> : <https://www.tldrlegal.com/license/mozilla-public-license-2-0-mpl-2>
+
+  - This one allows people to use your code in their commercial closed source projects. Just the copied and/or modified code portion must be publicly available or freely given when asked. This license also requires the modder to include a link to your repository so others can get the original software. They do not need to state any changes they made to your code. Server owners are allowed to modify the software as this license specifies that “distribution” is only about sending a copy of the software itself to another. See here for info about distribution: <https://www.mozilla.org/en-US/MPL/2.0/FAQ/#what-does-distribute-mean>
```

Comment:
For this and the other three licenses, the full name of the licenses ought to be added with the shorthand fort in parentheses: "Lesser GNU Public License", "Open Source License", "Apache License", "Mozilla Public License".

Additionally, it may make more sense to order these in increasing order of copyleftness: MIT, Apache, OSL, MPL, LGPLv3. (Based on a quick read-through of those summaries; I may be incorrect).

Reply:
I have them ordered in more popularity in the modding community

Reply:
Hmmm. Alright then, as long as we're clear why it's in that order.